tests/devices/test_default_qubit_jax.py

```diff
@@ -0,0 +1,462 @@
+
+        res = cost(params)
+        expected_cost = (
+            jnp.sin(lam) * jnp.sin(phi) - jnp.cos(theta) * jnp.cos(lam) * jnp.cos(phi)
+        ) ** 2
+        assert jnp.allclose(res, expected_cost, atol=tol, rtol=0)
+
+        res = jax.grad(cost)(params)
+        expected_grad = (
+            jnp.array(
+                [
+                    jnp.sin(theta) * jnp.cos(lam) * jnp.cos(phi),
+                    jnp.cos(theta) * jnp.cos(lam) * jnp.sin(phi) + jnp.sin(lam) * jnp.cos(phi),
+                    jnp.cos(theta) * jnp.sin(lam) * jnp.cos(phi) + jnp.cos(lam) * jnp.sin(phi),
+                ]
+            )
+            * 2
+            * (jnp.sin(lam) * jnp.sin(phi) - jnp.cos(theta) * jnp.cos(lam) * jnp.cos(phi))
+        )
+        assert jnp.allclose(res, expected_grad, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("interface", ["autograd", "tf", "torch"])
+    def test_error_backprop_wrong_interface(self, interface, tol):
+        """Tests that an error is raised if diff_method='backprop' but not using
+        the Jax interface"""
+        dev = qml.device("default.qubit.jax", wires=1)
+
+        def circuit(x, w=None):
+            qml.RZ(x, wires=w)
+            return qml.expval(qml.PauliX(w))
+
+        error_type = qml.QuantumFunctionError if qml.tape_mode_active() else ValueError
+        with pytest.raises(
+            error_type,
+            match="default.qubit.jax only supports diff_method='backprop' when using the jax interface",
+        ):
+            qml.qnode(dev, diff_method="backprop", interface=interface)(circuit)
+
+
+class TestHighLevelIntegration:
+    """Tests for integration with higher level components of PennyLane."""
+
+    def test_template_integration(self):
+        """Test that a PassthruQNode default.qubit.autograd works with templates."""
+        dev = qml.device("default.qubit.jax", wires=2)
+
+        @qml.qnode(dev, diff_method="backprop", interface="jax")
+        def circuit(weights):
+            qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        weights = jnp.array(qml.init.strong_ent_layers_normal(n_wires=2, n_layers=2))
+
+        grad = jax.grad(lambda a: circuit(a).reshape(()))(weights)
+        assert grad.shape == weights.shape
+
+    def test_qnode_collection_integration(self):
+        """Test that a PassthruQNode default.qubit.jax works with QNodeCollections."""
+        dev = qml.device("default.qubit.jax", wires=2)
+
+        def ansatz(weights, **kwargs):
+            qml.RX(weights[0], wires=0)
+            qml.RY(weights[1], wires=1)
+            qml.CNOT(wires=[0, 1])
+
+        obs_list = [qml.PauliX(0) @ qml.PauliY(1), qml.PauliZ(0), qml.PauliZ(0) @ qml.PauliZ(1)]
+        qnodes = qml.map(ansatz, obs_list, dev, interface="jax")
+        if qml.tape_mode_active():
+            pass  # TODO(chase): Figure out why map doesn't set the interface in tape mode.

```

Comment:
Check with @josh146 on this, the collections and associated functions are probably still not 100% edge-case free, and in need of a refresh

Reply:
Also, rather than passing, better to do a pytest skip?

Reply:
That's super weird. The `qml.map()` tests pass in tape mode now, and the entire function source code is mode-independent 🤔

Reply:
Yeah, it was really frustrating. I'll see if I can debug why that was happening.

Reply:
It's still happening, but only in tape mode.

```
>       assert qnodes.interface == "jax"
E       AssertionError: assert None == 'jax'
E        +  where None = <pennylane.collections.qnode_collection.QNodeCollection object at 0x7f2000605c70>.interface
```